src/middlewares/role.middleware.js

```diff
@@ -6,90 +6,88 @@ import { User, Blacklist } from '../database/models';
+    if (!token) return res.status(403).json({ message: 'user not logged in' });
+    try {
+        const blackListed = await Blacklist.findOne({ where: { token } });
+        if (blackListed) return res.status(401).json({ message: 'please login first' });
 
-    const decoded = decodeToken(token);
-    const freshUser = await User.findByPk(decoded.userId, {
-      include: 'Role'
-    });
-    req.user = freshUser;
-    next();
-  } catch (error) {
-    return res.status(401).json({ message: 'access denied' });
-  }
+        const decoded = decodeToken(token);
+        const freshUser = await User.findByPk(decoded.userId, {
+            include: 'Role'
+        });
+        req.user = freshUser;
+        next();
+    } catch (error) {
+        return res.status(401).json({ message: 'access denied' });
+    }
 };
 export const roles = (...roles) => {
-  return (req, res, next) => {
-    if (!roles.includes(req.user.dataValues.Role.dataValues.name)) {
-      return next(
-        res
-          .status(403)
-          .json({ message: 'you are not allowed to perform this action' })
-      );
-    }
-    next();
-  };
+    return (req, res, next) => {
+        if (!roles.includes(req.user.dataValues.Role.dataValues.name)) {
+            return next(
+                res
+                .status(403)
+                .json({ message: 'you are not allowed to perform this action' })
+            );
+        }
+        next();
+    };
 };
-export const TravelAdmin = async (req, res, next) => {
-  const token =
-    req.headers.authorization && req.headers.authorization.split(' ')[1];
-  let payload;
-  try {
-    payload = decodeToken(token);
-  } catch (error) {
-    return res.status(403).json({ message: 'user not logged in' });
-  }
+export const TravelAdmin = async(req, res, next) => {
+    const token =
+        req.headers.authorization && req.headers.authorization.split(' ')[1];
+    let payload;
+    try {
+        payload = decodeToken(token);
+    } catch (error) {
+        return res.status(403).json({ message: 'user not logged in' });
+    }
 
-  const user = await User.findOne({
-    where: { email: payload.email },
-    include: 'Role'
-  });
+    const user = await User.findOne({
+        where: { email: payload.email },
+        include: 'Role'
+    });
 
-  if (user.Role.name !== 'TRAVEL_ADMIN') {
-    return res.status(401).json({ message: 'only Travel Admin is allowed' });
-  }
+    if (user.Role.name !== 'TRAVEL_ADMIN') {
+        return res.status(401).json({ message: 'only Travel Admin is allowed' });
+    }
 
-  const { email, id, manager_id } = user;
+    const { email, id, manager_id } = user;
 
-  req.user = { email, id, manager_id };
-  next();
+    req.user = { email, id, manager_id };
+    next();
 };
 
-export const checkRoleSame = async (req, res, next) => {
-  const { email, role } = req.body;
+export const checkRoleSame = async(req, res, next) => {
```

Comment:
What about this middleware? Try to use meaningful name and make sure that you're returning appropriate status code